model/log.go

```diff
@@ -176,21 +176,21 @@ func GroupLines(lines []LogLine) ([]LogChunk, error) {
 	var currentChunk LogChunk
 
 	logChars := 0
-	for _, line := range lines {
-		if len(line.Msg) > maxLogChars {
+	for i := range lines {
+		if len(lines[i].Msg) > maxLogChars {
 			return nil, errors.New("Log line exceeded 4MB")
 		}
 
-		if len(line.Msg)+logChars > maxLogChars {
+		if len(lines[i].Msg)+logChars > maxLogChars {
 			logChars = 0
 			chunks = append(chunks, currentChunk)
 			currentChunk = LogChunk{}
 		}
 
-		logChars += len(line.Msg)
-		currentChunk.Lines = append(currentChunk.Lines, line)
-		if currentChunk.EarliestTime != nil && line.Time.Before(*currentChunk.EarliestTime) {
-			currentChunk.EarliestTime = &line.Time
+		logChars += len(lines[i].Msg)
```

Comment:
I'm not sure why we changed this to use the index instead?

Reply:
This was related to what you pointed out in [the next thread](https://github.com/evergreen-ci/logkeeper/pull/78#discussion_r936974048): we were capturing a pointer to the `line.Time` on each iteration of the loop. Since the iterator variable is always the same the pointers were always to the same thing so all the chunks had the earliest time of the last chunk.
I think the whole idea was unwise and opaque. I've removed it.